include/SFML/Graphics/Glsl.inl

```diff
@@ -57,15 +48,13 @@ struct Matrix
     ////////////////////////////////////////////////////////////
     /// \brief Construct from raw data
     ///
-    /// \param pointer Points to the beginning of an array that
-    ///                has the size of the matrix. The elements
-    ///                are copied to the instance.
+    /// \param data An array that has the size of the matrix.
+    ///             The elements are copied to the instance.
     ///
     ////////////////////////////////////////////////////////////
-    explicit Matrix(const float* pointer)
-    {
-        copyMatrix(pointer, Columns * Rows, array);
-    }
+    explicit Matrix(const std::array<float, Columns * Rows>& data)
+    : array{data}
+    {}
```

Comment:
This is less powerful than the previous API.

`const float*` allows for various APIs which don't use `std::array`. Not just C ones, even `std::vector`, or an `std::array` with more (unused) elements.

We could add a constructor overload, for symmetry reasons with `getMatrix()`.

Reply:
I put back the constructor for raw data.
I wonder if the overload with `std::array` is so useful. The good abstraction would be `std::span` but it comes with C++20.

Reply:
The constructor taking `std::array` seems useless or too specialized after all. I removed it.

Reply:
Following this line of thought, don't we also want to revert to returning `const float*` in `getMatrix()`?

By returning `const std::array&`, we allow precisely one implementation and that is having a field which is `std::array`. So it's almost as if we had the field public for read access.

Again, a `std::span` (C++20) would fit here, but `float*` is more generic. We just need to document how many elements are behind the pointer.

Reply:
The motivation for this PR was to return `const std::array&` to make writing unit tests simpler.
I am now not very sure it is the way to go. I agree it does not feel so elegant in `sf::Transform`'s interface. What we would need to have a nice interface is indeed `std::span`.

Now, I think we only need a testing infrastructure that can handle `const float*` arrays and switching to `std::array` bring little to no benefit.

Reply:
What if we just have a thin wrapper type around `const float*` that is used in tests?
Could even be `std::array<float, 16>` with an easy constructor function `toArray(floatPtr)` or so...

Reply:
I agree, I implemented this approach.